src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1206,6 +1263,51 @@ private ProcessStartInfo GetProcessStartInfo(bool redirectOutput, bool redirectE
             return startInfo;
         }
 
+        /// <summary>
+        /// Determine if we have a special file which will change the way native argument passing
+        /// is done on Windows. We use legacy behavior for cmd.exe, .bat, .cmd files.
+        /// </summary>
+        /// <param name="filePath">The file to use when checking how to pass arguments.</param>
+        /// <returns>A boolean indicating what passing style should be used.</returns>
+        private static bool UseLegacyPassingStyle(string filePath)
+        {
+            if (filePath == null || filePath == string.Empty)
+            {
+                return false;
+            }
+
+            string commandPath = filePath.ToLowerInvariant();
```

Comment:
No need to allocate. We can do `if (filePath.EndsWith(exception, StringComparison.OrdinalIgnoreVase))` in line 1302.